pkg/cluster/rook_ceph_test.go

```diff
@@ -3,11 +3,17 @@ package cluster
 import (
 	"context"
 	_ "embed"
+	"fmt"
+	"reflect"
 	"testing"
 
 	"github.com/blang/semver"
+	"github.com/golang/mock/gomock"
+	mock_k8s "github.com/replicatedhq/ekco/pkg/k8s/mock"
 	"github.com/replicatedhq/ekco/pkg/logger"
 	"github.com/replicatedhq/ekco/pkg/util"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookfake "github.com/rook/rook/pkg/client/clientset/versioned/fake"
```

Comment:
Not a blocker for we move forward here since we can discuss and improve it afterwords. 

However, we are using controller-runtime (https://github.com/rook/rook/blob/master/go.mod#L40) so ideally we should use ENV TEST instead of fake clients. It brings a poor experience and has many limitations, see:

- https://github.com/kubernetes-sigs/controller-runtime/blob/master/FAQ.md#q-wheres-the-fake-client--how-do-i-use-it and https://github.com/kubernetes-sigs/controller-runtime/pull/1031
- Also, controller-runtime has a old desired to deprecate/remove and it was not only removed because a lot of people has been using it:  https://github.com/kubernetes-sigs/controller-runtime/issues/768

Therefore, I'd like to suggest we use to do the tests ENV TEST ginkgo and gomega as it is done in other k8s projects such as controller-runtime as well. Following some examples:

- Kubebuilder: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3-with-deploy-image/controllers/memcached_controller_test.go
- Controller-Runtime: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3-with-deploy-image/controllers/memcached_controller_test.go

Reply:
Agreed but out of scope. Can you create a story in Replicated's internal Shortcut?

Reply:
I will do my best 👍